newa/__init__.py

```diff
@@ -129,6 +134,37 @@ class Event(Serializable):
     id: 'ErratumId'
 
 
+@define
+class ErrataTool:
+    """
+    Interface to Errata Tool instance
+
+    It exists only to fetch information about an erratum and its builds.
+    """
+
+    url : str = "https://errata.devel.redhat.com"
+
+    @staticmethod
+    def _request(url, max_attempts : int = 5, delay : int = 5) -> Any:
+        attempts = 0
+        while attempts < max_attempts:
+            r = requests.get(url, auth=HTTPKerberosAuth(delegate=True))
+            if r.status_code == 200:
+                return r.json()
+            else:
+                time.sleep(delay)
+                attempts += 1
+
+        raise Exception(f"GET request to {url} failed")
```

Comment:
* does it have to be a static method of `ErrataTool`? It seems to accept a generic URL, which may or may not point to ET, the code seems to not care. A helper function similar to `render_template()` would serve us as well.
* I'd recommend renaming `max_attempts` to `attempts`, then you can `while attempts` and `attempts -= 1` at the end of each loop. One less variable to play with.
* the `else` keyword is not needed, because the `if` branch ends with `return` - if the condition was true, `if` happens and function ends. If it was false, then the code after `if` is executed, `else` is then not needed.

Reply:
Yeah, I was on the edge on this and felt whole ErrataTool thing is something we don't really need and we can use the non-class methods instead. I'll move this one out and later we might drop whole ErrataTool thing if needed.